### DISCUSSION TITLE
Key on hyper layer triggers another layer instead of the layer action

### DISCUSSION THREAD

**Discussion by evan-liu** [evan-liu/karabiner.ts#142] [Q&A]
> Hey, apologies to revive this topic here but I have the opposite problem. 
> 
> When I am in one layer, I don't want another layer to get activated. All my layers are hyperLayer() and I define them in functions and call them in alphabetical order. I find that the layers first called are the ones that are able to use all alphabetical keys while the layers called after the first layer are unable to use the specific key used to activate the first layer.
> 
> For example, I have a yabaiLayer() that is Hyper + Y to activate the layer, then any key press is associated with an action. Before that, I have a raycastLayer() that's activated by Hyper + R. 
> 
> If I press, Hyper + Y + R, the action associated with R inside yabaiLayer() is never executed, rather raycastLayer() is stacked on top of yabaiLayer().

_Originally posted by @kayg04 in https://github.com/evan-liu/karabiner.ts/discussions/126#discussioncomment-11127743_

**Comment by evan-liu** [comment]
Hi, @kayg04. The original discussion is too long, so I am creating a new one. Can you post the code of one of your hyper-layers? I guess you may have "optional: any" modifiers.

**Reply by kaygdotorg** [reply]
hey, np, apologies for reviving the old discussion.

i added karabiner-ts config to my dotfiles, please check it out here: https://github.com/kayg04/dotfiles/blob/main/karabiner-ts/index.ts

**Reply by evan-liu** [reply]
Can you [fork the online editor](https://stackblitz.com/fork/github/evan-liu/karabiner.ts/tree/main/editor?file=rules.js) and create a minimal code to reproduce your problem?

**Reply by kaygdotorg** [reply]
Hey, here you go: https://stackblitz.com/edit/evan-liu-karabiner-ts-yjiyla?file=rules.js

**Reply by kaygdotorg** [reply]
I tested it. I cannot trigger number layer from within navigation layer but I can do the opposite so i am guessing it has to do with the order they are defined. Please let me know what you find. Thank you!

**Reply by evan-liu** [reply]
I've pushed a fix version, `1.30.1`. Can you give it a try to see if it solves your problem?

**Reply by kaygdotorg** [reply]
Yes, it does, thank you so much! 

Is there a way I can buy you a coffee or support your work in another way?

**Reply by evan-liu** [reply]
You are very welcome. I do not have anything set up, but thanks for offering ❤️
